src/main/resources/db/migration/0350/V376__VectorStoreLargerEmbedding.sql

```diff
@@ -0,0 +1,5 @@
+TRUNCATE vector_store;
+
+ALTER TABLE vector_store DROP COLUMN embedding;
+ALTER TABLE vector_store ADD COLUMN embedding halfvec(3072) NOT NULL;
```

Comment:
I'm assuming postgres doesn't like changing a `vector` column to `halfvec`, so you're dropping and adding instead of doing an `ALTER COLUMN`?

Reply:
Yeah, they're treated as incompatible data types. Plus the index type needs to change from `vector_cosine_ops` to `halfvec_cosine_ops` and index types don't change automatically when you alter a column type.